File: helloWorld.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <unistd.h>

char str[] = "HeLlO wOrLd";

char randomChar() {
    const char characters[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 ";
    int charactersLength = strlen(characters);
    return characters[rand() % charactersLength];
}

int main() {
    srand(time(NULL));
    char final[100] = "";
    int currentIndex = 0;

    while (final[currentIndex] != '\0' || str[currentIndex] != '\0') {
        char c = randomChar();
        final[currentIndex] = c;
        printf("%s\n", final);
        if (final[currentIndex] == str[currentIndex]) {
            currentIndex++;
        } else {
            final[currentIndex] = '\0';
        }
        usleep(10000);
    }

    return 0;
}

File: helloWorld.js
console.clear();
var str = "HeLlO wOrLd";

function randomString(length) {
    let result = '';
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 ';
    const charactersLength = characters.length;
    let counter = 0;
    while (counter < length) {
        result += characters.charAt(Math.floor(Math.random() * charactersLength));
        counter++;
    }
    return result;
}

let final = "";
let currentIndex = 0;

function generateNextCharacter() {
    if (final !== str) {
        let c = randomString(1);
        final += c;
        console.log(final);
        if (final[currentIndex] === str[currentIndex]) {
            currentIndex++;
        } else {
            final = final.slice(0, -1); // Cắt bỏ ký tự cuối cùng nếu không khớp
        }
        setTimeout(generateNextCharacter, 10); // Chờ trước khi tạo ký tự tiếp theo
    }
}

generateNextCharacter();
